python/TestHarness/util.py

```diff
@@ -119,38 +119,43 @@ def printResult(tester, result, timing, start, end, options, color=True):
     if m:
       any_match = True
       f_result += colorText(m.group(1), 'CYAN', **color_opts) + " "
-    # Color Exodiff or CVSdiff tests YELLOW
-    m = re.search('(FAILED \((?:EXODIFF|CSVDIFF)\))', result)
-    if m:
-      any_match = True
-      f_result += colorText(m.group(1), 'YELLOW', **color_opts)
+
+    ### Obtain test results using bucket status
+    if status:
+      # Color failing tests RED or YELLOW
+      if status == 'FAIL' or status == 'DIFF':
+        if status == 'DIFF':
+          color = 'YELLOW'
+        else:
+          color = 'RED'
+        any_match = True
+        f_result += colorText(result, color, **color_opts)
+
+      # Color Passing tests GREEN
+      elif status == 'PASS':
+        any_match = True
+        f_result += colorText(tester.getSuccessMessage(), 'GREEN', **color_opts)
+
+      # Color Pending tests CYAN
+      elif status == 'PENDING':
+        any_match = True
+        f_result += colorText(tester.getStatusMessage(), 'CYAN', **color_opts)
+
+      # Do not color skipped tests
+      elif status == 'SKIP':
+        any_match = False
+
+      # catch non-standard bucket status?
+      else:
+        any_match = False
```

Comment:
If we going to code up a big old case statement again, we still haven't succeeded with out cleanup. Perhaps we need even more information associated with all the statuses. We need to know what colors we want associated with the bucket statuses.